### ISSUE TITLE
Dynamic Dropdown throws Javascript Error on focus in Builder Backend

### ISSUE THREAD

**Issue by datune** [rainlab/builder-plugin#184]
Configuring a dropdown to retreive it's options from a Model method will, on focus of the dropdown in the builder plugin fields configuration window, throw a javascript error. 

Steps
- Add dropdown to your Model fields.yaml 
- Configure dropdown to retreive it's options using [one of the three dynamic methods](https://octobercms.com/docs/backend/forms#field-dropdown) available and add the appropiate method in your Model
- Save yaml file
- Open Builder in browser
- Navigate to the field configuration for your Model
- Click on your dropdown and you will see the error as shown below in your console

**fields.yaml**
```
        name_of_field:
            label: 'The Label'
            span: auto
            oc.commentPosition: ''
            type: dropdown
            options: listDropdownOptions
```

**Model method to retreive options**
```
public function listDropdownOptions($value, $formData)
{
        return [
          1 => 1,
          2 => 2,
          3 => 3,
          4 => 4,
          5 => 5,
          6 => 6,
          0 => 'Whatever....',
        ];
}
```

Build 425

**Error:**
```
storm-min.js?v425:5365 Uncaught Error: Object list value should be an object. Property: options
    at DictionaryEditor.BaseEditor.throwError (storm-min.js?v425:5365)
    at DictionaryEditor.setLinkText (storm-min.js?v425:6283)
    at DictionaryEditor.PopupBase.build (storm-min.js?v425:5583)
    at DictionaryEditor.BaseEditor.init (storm-min.js?v425:5349)
    at DictionaryEditor.init (storm-min.js?v425:6275)
    at DictionaryEditor.BaseEditor [as constructor] (storm-min.js?v425:5334)
    at DictionaryEditor.PopupBase [as constructor] (storm-min.js?v425:5574)
    at new DictionaryEditor (storm-min.js?v425:6267)
    at Surface.buildEditor (storm-min.js?v425:4764)
    at Surface.build (storm-min.js?v425:4672)
```
![image](https://user-images.githubusercontent.com/2123730/31537736-7be70dfc-b003-11e7-9541-dc837178643b.png)

Video:
[![Alt text](https://img.youtube.com/vi/qtmnfdIrN40/0.jpg)](https://youtu.be/qtmnfdIrN40)

**Comment by daftspunk** [comment]
Confirmed this bug. This is because the builder spec of a dropdown is different to the October spec. It wants an array and receives a string. It is not possible to set it as a string using builder, only manually.

We will fix this in the next major iteration of builder.

Thank you

**Comment by daftspunk** [comment]
Fixed as part of #372
